files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -310,6 +315,8 @@ public function validateMarkAsRead() {
 				$this->parameters['visitTime'] = TIME_NOW;
 			}
 		}
+
+		unset($this->parameters['userID']);
```

Comment:
The event runs *after* the `markAsRead()` call has completed, hence the `userID` parameter has already been disposed by that `unset()` call. Do not unset the user id there!